### DISCUSSION TITLE
Can you disable markdown list auto-completion?

### DISCUSSION THREAD

**Discussion by PathogenDavid** [community/community#23932] [New to GitHub]
<p>GitHub recently added support for <a href="https://github.blog/changelog/2020-12-15-markdown-list-syntax-now-autocompleted/" rel="noopener nofollow ugc">automatically extending markdown lists when you create a new line</a>.</p>
<p>Is there a way to disable this functionality? In general I dislike these sort of editor assistance features where things I didn’t type or commit are added for me. For example, I disable things like brace completion in all my IDEs and text editors. This feature is more annoying than helpful for me.</p>
<p>I know I’m weird, but I’m fighting against 20+ years of muscle memory. I’ve tried to get over it in the past, but at this point I’ve accepted these sort of features and my brain will never jive.</p>

**Comment by logankilpatrick** [comment]
<p>At this time, it is not a setting available to toggle. You can do what the blog post linked above mentioned though:</p>
<blockquote>
<p>If you’re at the end of your list, use <kbd>shift</kbd> + <kbd>enter</kbd> to skip the autocomplete and continue writing on the next line.</p>
</blockquote>

**Comment by jaydorsey** [comment]
<p>I’m in the same boat and would like a native option to disable this feature</p>
<p>It changes the default behavior of a textarea field which has behaved a certain way for many years.</p>
<p>If it was enabled for the custom list/checkboxes only that would totally make sense (because those are extra typing) but I make a lot of sub-lists and the auto-complete just gets in the way</p>
<p>This has been my day for the past week :rofl::</p>
<ul>
<li>Create my first bullet point, press enter
<ul>
<li>creates a new line/list item</li>
</ul>
</li>
<li>Press tab, because hey they’re creating a list item for me. Maybe they intercepted the tab/shift tab to indent easily
<ul>
<li>focus shifts to next element</li>
</ul>
</li>
<li>Press shift-tab
<ul>
<li>focus shifts back into the text area</li>
</ul>
</li>
<li>Backspace twice
<ul>
<li>deletes the <code>- </code>
</li>
</ul>
</li>
<li>Press space 3 times, and <code>-</code>
</li>
<li>Carry on with my typing</li>
</ul>

**Comment by jaydorsey** [comment, marked as answer]
<aside class="quote no-group" data-username="jaydorsey" data-post="3" data-topic="151277">
<div class="title">
<div class="quote-controls"></div>
 jaydorsey:</div>
<blockquote>
<p>would like a native option to disable this feature</p>
</blockquote>
</aside>
<p>I wrote something that I’m using with tampermonkey to disable this feature that seems to work OK:</p>
<aside class="onebox githubgist">
  <header class="source">
      <a href="https://gist.github.com/jaydorsey/d1c54ef5fa48668833851423341fbf6f" target="_blank" rel="noopener nofollow ugc">gist.github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://gist.github.com/jaydorsey/d1c54ef5fa48668833851423341fbf6f" target="_blank" rel="noopener nofollow ugc">https://gist.github.com/jaydorsey/d1c54ef5fa48668833851423341fbf6f</a></h4>
<h5>disable_github_list.js</h5>
<pre><code class="JavaScript">// ==UserScript==
// @name         Disable Github List Completion
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  Disables the behavior described in https://github.blog/changelog/2020-12-15-markdown-list-syntax-now-autocompleted/
// @author       Jay Dorsey
// @match        https://*.github.com/*
// @grant        window.focus
// ==/UserScript==
</code></pre>
This file has been truncated. <a href="https://gist.github.com/jaydorsey/d1c54ef5fa48668833851423341fbf6f" target="_blank" rel="noopener nofollow ugc">show original</a>

<p>
</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>I just guessed that the behavior was bound to a class name (I didn’t look at any of the JS itself), so I’m not sure what other behavior this might break. Please feel free to let me know if there’s a better solution</p>

**Comment by ferdnyc** [comment]
<p>It doesn’t annoy me enough to turn it off (or attempt to) yet, but I do find it kind of irksome only because it autoincrements the numeral each time.</p>
<p>I always write my Markdown lists in</p>
<pre><code class="lang-auto">1. Thing
1. Next thing
1. Another thing
</code></pre>
<p>form, so I can reorder the list at any time. Github’s editor is now fighting against my norms (or causing me to fight against it).</p>

**Comment by moorepants** [comment]
<p>Just another vote to disable this. Autocomplete features should be optional and opt in.</p>

**Comment by PathogenDavid** [comment]
<p>Thanks a lot, Jay! I finally got around to installing that workaround and it works great so far.</p>
<p>Barring an official solution I’ll accept that as a solution for now.</p>

**Comment by conroywhitney** [comment]
<p>Another vote for being able to toggle this. It’s like when Slack introduced WYSIWYG instead of letting us use markdown – but they finally added a toggle (thank goodness). The point is, some random PM/engineer’s decision to make things pretty ends up making my life harder. If you’re gonna mess with people’s workflows, give them the option to disagree with your opinion on what’s “best”.</p>

**Comment by markd2** [comment]
<p>+1 for letting us disable.  I have a very 1:1 correlation between what I type to what my mental model contains.  Editors “helpfully” inserting cause my mental model to get out of sync, leading to frustration and less joy when using the tools.</p>

**Comment by tedg** [comment]
<p>Yes I don’t like this functionality and would prefer to turn it off. When I create task lists in my PRs I would like to be able to copy the entirety of the line (e.g. <code>- [ ] Test Case 1</code>), press enter, and immediately paste it and be able to start editing the new todo item.</p>
<p>Since this feature has been released, when I do that workflow the new todo item looks like<br>
<code>- [ ] - [ ] Test Case 1</code>, so I have more work to do editing it. Please let us turn off this new feature.</p>

**Comment by indignant** [comment]
<p>I would also like to be able to turn this feature off. The “shift + enter” idea seemed like it would be a good temporary solution, but unfortunately it still indents on sub-lists.</p>

**Comment by reknock** [comment]
Has this problem not been solved yet?

**Reply by ferdnyc** [reply]
I don't get the impression that GitHub considers it a problem, so you shouldn't expect a solution from them, ever.

**Reply by mmrein** [reply]
As usual - Microsoft forcing features that no one aked for and users don't even want.

**Reply by ferdnyc** [reply]
Well, that's just false. Plenty of people asked for Markdown-composition streamlining features like list extension, and I personally _like_ the unordered list and checkbox list extension features. They're great, and very convenient! I would complain if they went away. I've even gotten used to ordered list extension, over time. It isn't my ideal version of the feature but it's way better than not having it at all.

Also, especially in 2020 GitHub was by all accounts still functioning _very_ independently from Microsoft, it's highly unlikely they had any involvement in something so nitty-gritty as the site's Markdown comment editor. Features like this were by GitHub and for GitHub, considering their web interface in general was one of the key advantages GH had over every other Git service provider, and a major factor in Microsoft valuing the acquisition so highly in the first place.

**Reply by mmrein** [reply]
We're not talking about unordered list or checkboxes in general, I like those too. We're talking specifically about autocomplete feature, which is annoying and should be optional.

**Comment by jamespharaoh** [comment]
I agree, this is super annoying and should be something you can disable, please.

**Reply by walchko** [reply]
This "feature" sucks
